daml/Marketplace/Trading/Confirmation/Model.daml

```diff
@@ -91,14 +93,35 @@ template Confirmation
               getAllocationAccount = getAccount accounts (.allocationAccount)
               getTradingAccount    = getAccount accounts (.tradingAccount)
 
+            let transferToFeeAccount : ContractId AssetDeposit -> Update (ContractId AssetDeposit)
+                transferToFeeAccount depositCid = do
+                  feeAccount <- (.feeAccount) . snd <$> fetchByKey @Order.FeeSchedule (operator, provider)
+                  exerciseByKey @AllocationAccountRule.T (getAllocationAccount party).id
+                      AllocationAccountRule.Transfer with transferTo = feeAccount; ..
+
             let (Order.Collateral depositCid) = order.collateral
             depositCid <- if isPartialFill
               then do
+                newOptExchangeFee <- case order.details.optExchangeFee of
+                  None               -> return None
+                  Some feeDepositCid -> do
+                    feeQuantity <- (.asset.quantity) <$> fetch feeDepositCid
+                    let feeToTake = roundBankers 2 $ feeQuantity * (execution.quantity / order.remainingQuantity)
+                    [feeDepositCid, remainingFeeDepositCid] <- exercise feeDepositCid AssetDeposit_Split with quantities = [feeToTake]
+                    transferToFeeAccount feeDepositCid
+                    return $ Some remainingFeeDepositCid
+
                 (depositCid :: depositRemainingCid :: _) <- exercise depositCid AssetDeposit_Split with ..
-                create order with executions, remainingQuantity, collateral = Order.Collateral depositRemainingCid, status = Order.PartiallyExecuted
+                create order with
+                  executions, remainingQuantity
+                  collateral = Order.Collateral depositRemainingCid, status = Order.PartiallyExecuted
+                  details = order.details with optExchangeFee = newOptExchangeFee
                 pure depositCid
               else do
-                create order with executions, remainingQuantity, status = Order.FullyExecuted
+                whenSome order.details.optExchangeFee (void . transferToFeeAccount)
+                create order with
+                  executions, remainingQuantity
+                  status = Order.FullyExecuted, details = order.details with optExchangeFee = None
```

Comment:
So we take the `Fees` on every execution received for an order ? Ie, we take them _in real time_ regardless if the execution has settled or not between participants ? If so, was that the requirement from Eric ? 

I'm just wondering if the payment of the fee should be part of the settlement process. Ie, we would exchange the assets between the holders and pay the exchange the fees as part of one atomic transaction.

Reply:
The payment of the fee is part of the settlement process - this is just locking the fee in the `AllocationAccount` to prevent it from being used elsewhere (similar to orders).

Reply:
I thought these two lines did an actual transfer of the fee to the exchange or am I mistaken :

`transferToFeeAccount feeDepositCid`
`return $ Some remainingFeeDepositCid`

and 

`whenSome order.details.optExchangeFee (void . transferToFeeAccount)`

Reply:
Ah, I apologize, I thought this was in `Trading.Service`! Is the idea to move this to `SettlementInstruction` then?

Reply:
Moving the transaction to `SettlementInstruction` worked well, see: https://github.com/digital-asset/da-marketplace/pull/387/commits/389532d304196c899a525b39569c210f51c8ea4a

Reply:
Awesome, thanks @alexgraham-da 👍